sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -112,6 +112,12 @@ private[sql] class DataFrameImpl protected[sql](
 
   override def printSchema(): Unit = println(schema.treeString)
 
+  override def printPlan(extended: Boolean): Unit = {
+    ExplainCommand(
+      logicalPlan,
+      extended = extended).rdd.collect().map(r => println(r.getString(0)))
```

Comment:
Does this run a job?  Why aren't you just calling `collect()` so that it goes through the `executeCollect()` code path?